src/GslibIO.jl

```diff
@@ -52,28 +80,51 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
-
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  dataspec = parse_legacy(filename)                     
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # handle missing values
+  replace!(dataspec.data, na=>NaN)
 
-    # read property values
-    X = readdlm(fs)
+  # create data dictionary
+  table = (; zip(dataspec.varnames, eachcol(dataspec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # handle missing values
-    replace!(X, na=>NaN)
+  georef(table, domain)
+end
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    georef(data, domain)
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  dataspec = parse_legacy(filename)
```

Comment:
Rename `dataspec` to `spec` to reduce the verbosity in the following lines of the algorithm.